third_party/skia/platform_view_rtree.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2012 Google Inc.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#ifndef SKIA_PLATFORM_VIEW_RTREE_H_
+#define SKIA_PLATFORM_VIEW_RTREE_H_
+
+#include <list>
+
+#include "third_party/skia/include/core/SkBBHFactory.h"
+#include "third_party/skia/include/core/SkTypes.h"
+
+/**
+ * An R-Tree implementation. In short, it is a balanced n-ary tree containing a hierarchy of
+ * bounding rectangles.
+ *
+ * It only supports bulk-loading, i.e. creation from a batch of bounding rectangles.
+ * This performs a bottom-up bulk load using the STR (sort-tile-recursive) algorithm.
+ *
+ * TODO: Experiment with other bulk-load algorithms (in particular the Hilbert pack variant,
+ * which groups rects by position on the Hilbert curve, is probably worth a look). There also
+ * exist top-down bulk load variants (VAMSplit, TopDownGreedy, etc).
+ *
+ * For more details see:
+ *
+ *  Beckmann, N.; Kriegel, H. P.; Schneider, R.; Seeger, B. (1990). "The R*-tree:
+ *      an efficient and robust access method for points and rectangles"
+ *
+ * The original source code can be found on
+ * https://github.com/google/skia/blob/508fd32091afe334d4e1dd6cdaa63168a53acb26/src/core/SkRTree.h
+ *
+ * This copy includes a searchRects method.
+ */
+class PlatformViewRTree : public SkBBoxHierarchy {
+public:
+    PlatformViewRTree();
+
+    void insert(const SkRect[], const SkBBoxHierarchy::Metadata[], int N) override;
+    void insert(const SkRect[], int N) override;
+    void search(const SkRect& query, std::vector<int>* results) const override;
+
+    // Finds the rects in the tree that represent drawing operations and intersect
+    // with the query rect.
+    //
+    // When two rects intersect with each other, they are joined into a single rect
+    // which also intersects with the query rect. In other words, the bounds of each
+    // rect in the result list are mutually exclusive.
+    //
+    // Since this method is used when compositing platform views, the rects in the
+    // result list represent UIViews that are composed on top of the platform view.
+    //
+    // However, Skia uses this tree to tracks operations that don't have any context
```

Comment:
I'm not sure I understand the message this paragraph is trying to convey to someone coming here to read about searchRects?